sdk/src/metrics/metric_reader.cc

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ENABLE_METRICS_PREVIEW
+#  include "opentelemetry/sdk/metrics/metric_reader.h"
+#  include "opentelemetry/sdk/metrics/export/metric_producer.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+
+MetricReader::MetricReader(AggregationTemporarily aggregation_temporarily)
+    : aggregation_temporarily_(aggregation_temporarily)
+{}
+
+void MetricReader::SetMetricProducer(MetricProducer *metric_producer)
+{
+  metric_producer_ = metric_producer;
+}
+
+AggregationTemporarily MetricReader::GetAggregationTemporarily() const noexcept
+{
+  return aggregation_temporarily_;
+}
+
+bool MetricReader::Collect(nostd::function_ref<bool(MetricData)> callback) noexcept
+{
+  if (!metric_producer_)
+  {
+    OTEL_INTERNAL_LOG_WARN(
+        "MetricReader::Collect Cannot invoke Collect(). No MetricProducer registered for "
+        "collection!")
+  }
+  if (shutdown_)
```

Comment:
`shutdown_` can be accessed in multiple threads?

Reply:
Yes, good point. Have protected it now.